# Day-1

In [22]:
import pandas as pd 
import numpy as np

### Load train.csv dataset and get the age column, then drop the null values

In [ ]:
data = pd.read_csv("../../week1/day4/train.csv")
age = data["Age"]
age=age.dropna()
age.info()

<class 'pandas.Series'>
Index: 714 entries, 803 to 630
Series name: Age
Non-Null Count  Dtype  
--------------  -----  
714 non-null    float64
dtypes: float64(1)
memory usage: 11.2 KB


### Calcualte some statistics the intership has asked

In [ ]:
mean=np.mean(age)
median=np.median(age)

v,c= np.unique(age,return_counts=True)
mode= v[np.argmax(c)]
std = np.std(age)
q1,q3 = np.percentile(age,[25,75])
iqr= q3-q1

print("Mean:", mean)
print("Median:", median)
print("Mode:", mode)
print("Standard Deviation:", std)
print("IQR:", iqr)


Mean: 29.69911764705882
Median: 28.0
Mode: 24.0
Standard Deviation: 14.516321150817317
IQR: 17.875


### Step 3: Identify whether the mean or median better represents a 'typical' value, and justify the choice.

I think the median gives a better idea of the typical passenger's age. The median age is 28, while the mean is about 29.7. The mean is slightly higher because some older passengers increase the average. Since the median is less affected by outliers, it represents the typical age more accurately.

### What Ive done in the pre steps:
1. I loaded the **train.csv** that is for the titanic dataset 

2. **step1** asked to laod numirc column so I get the column **Age** as data['Age'], and then drop the null values 

3. **step2** asked to calculate some statistics, all of them was to be honest easy to do, only the **IQR** it was new to me 
so I asked **GPT** how to calculate and explains to me **why** we calculate it and what the **result means**  

4. **step3** was a question wether the **mean** or **median** better, my answer was the median better, cuz it gives a better idea of the typical passenger's age